business/checkers/k8shttproutes/no_k8sgateway_checker.go

```diff
@@ -0,0 +1,55 @@
+package k8shttproutes
+
+import (
+	"fmt"
+
+	k8s_networking_v1alpha2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+
+	"github.com/kiali/kiali/kubernetes"
+	"github.com/kiali/kiali/models"
+)
+
+type NoK8sGatewayChecker struct {
+	K8sHTTPRoute *k8s_networking_v1alpha2.HTTPRoute
+	GatewayNames map[string]struct{}
+}
+
+// Check validates that the HTTPRoute is pointing to an existing Gateway
+func (s NoK8sGatewayChecker) Check() ([]*models.IstioCheck, bool) {
+	validations := make([]*models.IstioCheck, 0)
+
+	valid := s.ValidateHTTPRouteGateways(&validations)
+
+	return validations, valid
+}
+
+// ValidateHTTPRouteGateways checks all HTTPRoute gateways and checks that they're found from the given list of gatewayNames. Also return index of missing gatways to show clearer error path in editor
+func (s NoK8sGatewayChecker) ValidateHTTPRouteGateways(validations *[]*models.IstioCheck) bool {
+	valid := true
+
+	if len(s.K8sHTTPRoute.Spec.ParentRefs) > 0 {
+		for index, parentRef := range s.K8sHTTPRoute.Spec.ParentRefs {
+			if string(parentRef.Name) != "" && string(*parentRef.Kind) == kubernetes.K8sActualGatewayType && string(*parentRef.Group) == kubernetes.K8sNetworkingGroupVersionV1Beta1.Group {
+				namespace := s.K8sHTTPRoute.Namespace
+				if string(*parentRef.Namespace) != "" {
```

Comment:
It is possible to add an additional check? 

`if parentRef.Namespace != nil && string(*parentRef.Namespace) != "" {`

To avoid error: 
panic: runtime error: invalid memory address or nil pointer dereference

Reply:
good catch, thank you :+1: